network-config-analyzer/NetworkConfigQuery.py

```diff
@@ -531,39 +532,105 @@ class ConnectivityMapQuery(NetworkConfigQuery):
     def get_supported_output_formats():
         return {'txt', 'yaml', 'csv', 'md', 'dot'}
 
+    def is_in_subset(self, peer):
+        """
+        returns indication if the peer element is in the defined subset
+        Please note: Subset is a sort of a filter. It filters out elements which WERE DEFINED and do not match the settings.
+        Thus, the function returns True if no subset was defined
+        since, in this case, the subset is infinite and everything is "in the subset"
+        :param peer: peer element to filter (currently supports only Pods)
+        :return:
+        """
+        # if subset restrictions were not defined at all, everything is in the subset
+        if not self.output_config.subset:
+            return True
+
+        subset = self.output_config.subset
+        # filter by namespace
+        if isinstance(peer, Peer) and subset.get('namespace_subset') \
+                and str(peer.namespace) in str(subset['namespace_subset']).split(','):
+            return True
+
+        # filter by deployment
+        if isinstance(peer, Pod) and subset.get('deployment_subset') and peer.owner_name:
+            dep_names = str(subset.get('deployment_subset')).split(',')
+            for dep_name in dep_names:
+                if '/' in dep_name:
+                    dep_name_for_comp = peer.workload_name
+                else:
+                    dep_name_for_comp = peer.owner_name
+                if dep_name in dep_name_for_comp:
+                    return True
+
+        # filter by label
+        if isinstance(peer, Peer) and subset.get('label_subset') and peer.labels:
+            # go over the labels and see if all of them are defined
+            for single_label_subset in list(subset['label_subset']):
+                if isinstance(single_label_subset, dict):
+                    if self.are_labels_all_included(single_label_subset, peer.labels):
+                        return True
+                else:
+                    if self.are_labels_all_included(subset['label_subset'], peer.labels):
+                        return True
+                    break
+
+        return False
+
+    @staticmethod
+    def are_labels_all_included(target_labels, pool_labels):
+        for key in target_labels:
+            if (key, target_labels[key]) not in pool_labels.items():
+                return False
```

Comment:
```suggestion
        for key, val in target_labels.items():
            if pool_labels.get(key) != val:
                return False
```
This is far more efficient: O(n) vs. O(n^2)

Reply:
Done